src/Sil/Component/Invoice/Model/Invoice.php

```diff
@@ -0,0 +1,590 @@
+     * taxes.
+     *
+     * @var taxes
+     */
+    protected $taxes;
+
+    /**
+     * subtotal.
+     *
+     * @var float
+     */
+    protected $subTotal;
+
+    /**
+     * grand total.
+     *
+     * @var float
+     */
+    protected $grandTotal;
+
+    /**
+     * paid amount.
+     *
+     * @var float
+     */
+    protected $paidAmount;
+
+    /**
+     * due amount.
+     *
+     * @var float
+     */
+    protected $dueAmount;
+
+    /**
+     * payment terms.
+     *
+     * @var string
+     */
+    protected $paymentTerms;
+
+    /**
+     * created at.
+     *
+     * @var DateTime
+     */
+    protected $createdAt;
+
+    /**
+     * customer.
+     *
+     * @var CustomerInfoInterface
+     */
+    protected $customer;
+
+    /**
+     * invoice items.
+     *
+     * @var array|InvoiceItemInterface[]
+     */
+    protected $items;
+
+    /**
+     * invoice adjustments.
+     *
+     * @var array|InvoiceAdjustmentInterface[]
+     */
+    protected $adjustments;
+
+    /**
+     * @param string                $code
+     * @param DateTime              $date
+     * @param CurrencyInterface     $currency
+     * @param float                 $taxRate
+     * @param string                $sellerInfo
+     * @param DateTime              $dueDate
+     * @param float                 $taxes
+     * @param float                 $subTotal
+     * @param float                 $grandTotal
+     * @param float                 $paidAmount
+     * @param float                 $dueAmount
+     * @param string                $paymentTerms
+     * @param CustomerInfoInterface $customer
+     */
+    public function __construct(
+        string $code,
+        \DateTime $date,
+        CurrencyInterface $currency,
+        float $taxRate,
+        string $sellerInfo,
+        \DateTime $dueDate,
+        float $taxes,
+        float $subTotal,
+        float $grandTotal,
+        float $paidAmount,
+        float $dueAmount,
+        string $paymentTerms,
+        CustomerInfoInterface $customer
+    ) {
+        $this->code = $code;
+        $this->date = $date;
+        $this->currency = $currency;
+        $this->taxRate = $taxRate;
+        $this->sellerInfo = $sellerInfo;
+        $this->dueDate = $dueDate;
+        $this->taxes = $taxes;
+        $this->subTotal = $subTotal;
+        $this->grandTotal = $grandTotal;
+        $this->paidAmount = $paidAmount;
+        $this->dueAmount = $dueAmount;
+        $this->paymentTerms = $paymentTerms;
+        $this->customer = $customer;
+        $this->createdAt = new \DateTime();
+        $this->items = new ArrayCollection();
+        $this->adjustments = new ArrayCollection();
+    }
+
+    /**
+     * Get the value of code.
+     *
+     * @return string
+     */
+    public function getCode(): string
+    {
+        return $this->code;
+    }
+
+    /**
+     * Set the value of code.
+     *
+     * @param string $code
+     */
+    public function setCode(string $code): void
```

Comment:
All setters must be checked because, for some of them, they does not need to be defined (e.g.: for mandatory fields used in constructor)